### PR TITLE
Extension installation overlay

### PR DESCRIPTION
**Pull request by jacogr**
TODO: Test once extension is published with new/correct extension URL.

Displays application prompt for extension installation. (Closes https://github.com/ethcore/parity/issues/4300)

Based off #4412 with changes -

- Doesn't display on hidden Web view, rather displays as part of application
- Proper store set for backing the component, rather than loose-standing calls

Future work -

- Create store & component tests
- Integrate prompt with Home view once available

parity 2017-02-03 15-43-08

### CODE REVIEW

File: js/src/views/Application/Extension/extension.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+import Store from './store';
+import styles from './extension.css';
+
+@observer
+export default class Extension extends Component {
+  store = new Store();
+
+  render () {
+    const { shouldShowWarning } = this.store;
```
Comment (ngotchac) [inline js/src/views/Application/Extension/extension.js]: Why not `showWarning` ? Being polite is good though

File: js/src/views/Application/Extension/extension.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    return (
+      <div className={ styles.body }>
+        <CloseIcon
+          className={ styles.close }
+          onClick={ this.onClose }
+        />
+        <p>
+          <FormattedMessage
+            id='extension.intro'
+            defaultMessage='Parity now has an extension available for Chrome that allows the viewing of Ethereum identities and the safe browsing of Ethereum-enabled distributed applications. It is hightly recommended that you install this extension to further enhance your Parity experience.'
```
Comment (ngotchac) [inline js/src/views/Application/Extension/extension.js]: For now, the published extension won't have the augmentation part. Thus we can get rid of `the viewing of Ethereum identities`

File: js/src/views/Application/Extension/store.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  }
+
+  @action hideWarning = (sleep = A_DAY) => {
+    this.nextDisplay = Date.now() + sleep;
+    store.set(NEXT_DISPLAY, this.nextDisplay);
+  }
+
+  @action testInstall = () => {
+    this.shouldInstall = this.readStatus();
+    console.log('testInstall', this.shouldInstall);
```
Comment (ngotchac) [inline js/src/views/Application/Extension/store.js]: Not sure if logging is needed here. We already have too many logs by default. If needed, could use the new `logger` feature

File: js/src/views/Application/Extension/store.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  }
+
+  readStatus = () => {
+    const hasExtension = Symbol.for('parity.extension') in window;
+    const ua = browser.analyze(navigator.userAgent || '');
+
+    console.log('readStatus', hasExtension, ua);
```
Comment (ngotchac) [inline js/src/views/Application/Extension/store.js]: Same as above

File: js/src/views/Application/Extension/store.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+      }
+    })
+    .catch((error) => {
+      console.warn('Unable to perform direct install', error);
+      window.open(EXTENSION_PAGE, '_blank');
+
+      this.hideWarning(TEN_MINUTES + A_MINUTE);
+      setTimeout(() => {
+        this.testInstall();
+      }, TEN_MINUTES);
```
Comment (ngotchac) [inline js/src/views/Application/Extension/store.js]: Shouldn't we have a way of closing the Warning only when the extension gets installed ? It seems bizarre to close it even if the installation failed (if the user clicked on `install` not on purpose, he can click on close).

Also, I'm not sure that the extension would actually be detected without a page refresh by just looking at the window object... To be verified

File: js/src/views/Application/Extension/store.js (hunk 3 above)
Comment (jacogr) [inline js/src/views/Application/Extension/store.js]: You are right, that one is a misplaced debugging comment. Thanks.

File: js/src/views/Application/Extension/store.js (hunk 4 above)
Comment (jacogr) [inline js/src/views/Application/Extension/store.js]: Same as above as well. Sorry.

File: js/src/views/Application/Extension/store.js (hunk 5 above)
Comment (jacogr) [inline js/src/views/Application/Extension/store.js]: Yes, 100% - would need to test this, updating the URL now to run through.

File: js/src/views/Application/Extension/extension.js [hunk 6]
```diff
@@ -44,7 +44,7 @@ export default class Extension extends Component {
         <p>
           <FormattedMessage
             id='extension.intro'
-            defaultMessage='Parity now has an extension available for Chrome that allows the viewing of Ethereum identities and the safe browsing of Ethereum-enabled distributed applications. It is hightly recommended that you install this extension to further enhance your Parity experience.'
+            defaultMessage='Parity now has an extension available for Chrome that allows for the safe browsing of Ethereum-enabled distributed applications. It is hightly recommended that you install this extension to further enhance your Parity experience.'
```
Comment (ngotchac) [inline js/src/views/Application/Extension/extension.js]: 1. "it is *highly* recommended"
2. could get rid of "allows *for the* safe"

Comment (jacogr) [inline js/src/views/Application/Extension/extension.js]: Erk. Damnit.

**Comment by ngotchac** [comment]
Works fine, but I think we want to use the inline install : https://support.google.com/webmasters/answer/34592?hl=en

@tomusdrw Localhosts sites must be added to the extension config in the Chrome Store
There should be at least (127.0.0.1/localhost):300 and (127.0.0.1/localhost):8180

**Comment by jacogr** [comment]
*.web3.site is added to webmaster tools. However 1.5.1 doesn't have full support for it (yet). (And even in those cases we are not yet re-directing people, and cannot redirect all)

Inline install does run through, obviously just pops up.

For 1.5.1, fine as it, works, the extra auto-browser window popup is not the end of the world.

**Comment by ngotchac** [comment]
Sure it's not the end of the world, but if it's only a one/few-line(s) to add to the config of the extension, I don't see why we shouldn't do it

**Comment by jacogr** [comment]
100%, will check once the beta is build in a short while, hopefully the config is updated and it is slightly smoother. As it stands, no reason not to go. (Unless that config change breaks the process in some way)

**Comment by jacogr** [comment]
Merging as-is so testing with beta build can be done.

**Comment by jacogr** [comment]
s/grumble/looksgood/ (non code related concerns)

**Pull request merged**